Added support for register-with-taints setting #93608

kshitijcode opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #93608

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**: 
Yes

<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

ymmt2005 committed:
do not create StatefulSet pods when PVC is being deleted

Pod with PVC will not be scheduled if the PVC is being deleted.
This can happen when the PVC has finalizers of storage plugins.

Such a pod becomes pending.  Unfortunately, after the finalizer
finishes and PVC is deleted, the pod remains pending forever.
The StatefulSet controller does nothing for this pending pod.

This commit prevents the StatefulSet controller from creating
such pods when PVC is to be deleted.

codenrhoden committed:
Extract pkg/util/mount and drop BUILD

Preserving git history.

codenrhoden committed:
Update doc.go to show k8s.io/utils

codenrhoden committed:
Fix golint errors

Add mount package to hack/.golint_failures to ignore stuttering errors.

Commit:
use xfs_repair to check and repair xfs filesystem

Signed-off-by: Lou <luogj@cn.ibm.com>

SataQiu committed:
feature: implement Output method for FakeCmd

mvisonneau committed:
Validate the existence of filesystem before attempting to mount it (linux)

gnufied committed:
Return typed error when Mount Fails

Mount can fail for a variety of reasons and caller might want to know
why mount failed. Using untyped string based error does not provide
enough granularity to make that verification.

Commit:
update after review

Signed-off-by: Lou <luogj@cn.ibm.com>

dims committed:
Split MajorMinor into two fields

this reflects what is in the docker/docker/pkg/mount API. makes it
easier to port additional code to out API.

davidz627 committed:
Add more detailed error output when disk formatting fails

davidz627 committed:
FormatAndMount unit test only checks for MountErrorValue now and closed gaps for some error values

saad-ali committed:
Introduce paramater for sensitive mount options.

Introduce optional sensitiveOptions parameter to allow sensitive mount
options to be passed in a separate parameter from the normal mount
options and ensures the sensitiveOptions are never logged.

BenTheElder committed:
fix cluster/log-dump/log-dump.sh shellcheck failures

saad-ali committed:
Fix mount_windows build error

tamalsaha committed:
Remove --include-uninitialized flag documentation

This flag does not exist anymore.

kvaps committed:
Fix subPath mountpint check

mboersma committed:
Fix windows MountSensitive error

rphillips committed:
drain: eviction creates in a deleting namespace will throw a forbidden error

we can safely ignore this error since a deleting namespace will delete
all pods

gnufied committed:
Revert xfs_repair fix

dims committed:
Switch to klog v2

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

dims committed:
fix bad spelling

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

dims committed:
ignore golint for some stutter that we want to keep as-is

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

joakimr-axis committed:
Fix shellcheck w/e in cluster/gce/gci/configure-helper.sh

Change-Id: Ic8fca2509a7cb07f4170eaf25a878036d18ba51c
Signed-off-by: Joakim Roubert <joakimr@axis.com>

joakimr-axis committed:
Code review update

Change-Id: I384a73efe995c529fb4b3636cb9639eafb90787f
Signed-off-by: Joakim Roubert <joakimr@axis.com>

Commit:
cleanup: no need nil check before range

fedepaol committed:
Add tests for SCTP network policies.

Signed-off-by: Federico Paolinelli <fpaoline@redhat.com>

Commit:
Avoid using socket for hints

Sockets don't affect performance as NUMA node does, since NUMA
node has dedicated memory controller, but socket it's physical
extension point.
Socket it's only cpu specific thing and it's strange to merge bitmask of
deviceplugin's and cpu manager, when cpu manager takes into account
socket.

Signed-off-by: Alexey Perevalov <alexey.perevalov@huawei.com>

Commit:
move to cadvisor.MachineInfo

This patch removes GetNUMANodeInfo, cadvisor.MachineInfo will be used
instead of it. GetNUMANodeInfo was introduced due to difference of meaning of
MachineInfo.Topology. On the arm it was NUMA nodes, but on the x86 it
represents sockets (since reading from /proc/cpuinfo). Now it unified
and MachineInfo.Topology represents NUMA node.

Signed-off-by: Alexey Perevalov <alexey.perevalov@huawei.com>

joakimr-axis committed:
Update after code review

Simplified local variable declaration as suggested by phenixblue.

Signed-off-by: Joakim Roubert <joakim.roubert@axis.com>

joakimr-axis committed:
Updates after code review

Add double quotes at assignments as requested by phenixblue.

Signed-off-by: Joakim Roubert <joakim.roubert@axis.com>

joakimr-axis committed:
Mitigate newly added shellcheck issues

Issues not present when the original patch was created have now also
been fixed.

Signed-off-by: Joakim Roubert <joakim.roubert@axis.com>

joakimr-axis committed:
Update cluster/gce/gci/configure-helper.sh

Co-authored-by: Aaron Crickenberger <spiffxp@google.com>

joakimr-axis committed:
Update cluster/gce/gci/configure-helper.sh

Co-authored-by: Aaron Crickenberger <spiffxp@google.com>

joakimr-axis committed:
Update cluster/gce/gci/configure-helper.sh

Co-authored-by: Aaron Crickenberger <spiffxp@google.com>

joakimr-axis committed:
Update cluster/gce/gci/configure-helper.sh

Co-authored-by: Aaron Crickenberger <spiffxp@google.com>

joakimr-axis committed:
Updates after review

Signed-off-by: Joakim Roubert <joakim.roubert@axis.com>

jingxu97 committed:
Remove driver letter assignment during volume format

This PR removes the driver letter assignment during volume format and
mount because driver letter might run out and cause issues during mount.
Intead, it uses volume id to mount the target dir.

jingxu97 committed:
Remove driver letter assignment during volume format

This PR removes the driver letter assignment during volume format and
mount because driver letter might run out and cause issues during mount.
Intead, it uses volume id to mount the target dir.

zvier committed:
fix golint errors for pkg/kubelet/dockershim

Signed-off-by: zvier <zvier20@gmail.com>

knabben committed:
Adding tests for Kubelet pod update functions

sxllwx committed:
lock-free broadcaster, use chan to ensure thread safety

andyzhangx committed:
fix: remove unnecessary readlink check in IsLikelyNotMountPoint on Windows

fix build failure

fix comments

fix build failure

fix comments

andyzhangx committed:
chore: add more logging for mklink on Windows

use volumeID var

MikeSpreitzer committed:
Remove contemplation of invariant violations from delta_fifo.go

Some comments and code incorrectly contemplated violating the
invariant that a keys is in `f.items` if and only if it is in
`f.queue`.

Also fixed up some comment wording.

MHBauer committed:
basic regression test of runDockershim

 - added basic regression test to ensure an error is raised in the
   case of an unconfigured runtime, and the case of asking for a docker
   runtime when compiled dockerless

Signed-off-by: Morgan Bauer <mbauer@us.ibm.com>

YuikoTakada committed:
Fix broken format in CHANGELOG-1.18

adrian committed:
Add unit test to kubectl/cmd/run

This test verifies that adding the "expose" parameter results in a
service being created for the pod.

zhijianli88 committed:
cleanup tempfiles in unit test

Signed-off-by: Li Zhijian <lizhijian@cn.fujitsu.com>

snowplayfire committed:
add line break when describe hpa

jingxu97 committed:
Revert "Merge pull request #166 from jingxu97/May/drivename"

This reverts commit 278ece378a5054f16a07622b51ddaf82b328d6e6, reversing
changes made to 2df71ebbae66f39338aed4cd0bb82d2212ee33cc.

SergeyKanzhelev committed:
remove stale TODO after this PR: #92204

joakimr-axis committed:
Add fix for run-kube-controller-manager-as-non-root

Signed-off-by: Joakim Roubert <joakim.roubert@axis.com>

kishorj committed:
AWS cloudprovider allow nlb-ip and external type lbs to be managed externally

panpan0000 committed:
Fix static check failures for staging/src/k8s.io/apiserver/pkg/server/healthz/healthz_test.go

Commit:
add defer

joakimr-axis committed:
No quotes needed/wanted for CURL_RETRY_CONNREFUSED

Signed-off-by: Joakim Roubert <joakim.roubert@axis.com>

joakimr-axis committed:
Remove previously added '' no longer needed

Adapt to changes on master since the first commit here.

Signed-off-by: Joakim Roubert <joakim.roubert@axis.com>

BenTheElder committed:
disable goffuzz in binaries

joakimr-axis committed:
Sync with master

Add fixes for newly added code.

Signed-off-by: Joakim Roubert <joakim.roubert@axis.com>

joakimr-axis committed:
Review update

Signed-off-by: Joakim Roubert <joakim.roubert@axis.com>

joakimr-axis committed:
Find what fails pull-kubernetes-e2e-gce-ubuntu-containerd

Change-Id: I7919d03926880cd9c93c61a07ada645ebfe32a89
Signed-off-by: Joakim Roubert <joakim.roubert@axis.com>

kensipe committed:
fix S1002 omit comp to bool

Signed-off-by: Ken Sipe <kensipe@gmail.com>

kensipe committed:
fix S1008 simplify bool returns

Signed-off-by: Ken Sipe <kensipe@gmail.com>

kensipe committed:
fix S1000 simplify ch switch cases

Signed-off-by: Ken Sipe <kensipe@gmail.com>

kensipe committed:
fix S1000 simplify ch switch cases

Signed-off-by: Ken Sipe <kensipe@gmail.com>

kensipe committed:
changes in imports was unintentional

Signed-off-by: Ken Sipe <kensipe@gmail.com>

phinphing committed:
fix typo in runtime/interfaces.go

SergeyKanzhelev committed:
SIG node owner files clean up

gongguan committed:
skip mismatched webhookAccessor and object

farah committed:
Add unittest coverage for boottime_util_linux.go

MikeSpreitzer committed:
Repair instead of panic when data corruption detected in DeltaFIFO

andyxning committed:
code clean for podgc

chendave committed:
Change the node name from "machine" to "node"

Latest change on master rename the node name from "machine" to "node"
but haven't update all the affected code, which causes some of testcases
invalid.

Signed-off-by: Dave Chen <dave.chen@arm.com>

Commit:
Remove DeprecatedMightBeMasterNode()

This removes DeprecatedGetMasterAndWorkerNodes() usage from vsphere e2e
test as deprecated function cleanup.
Then all callers of DeprecatedMightBeMasterNode() have been removed.
So this removes DeprecatedMightBeMasterNode() itself also.

jvanz committed:
cleanup: decrease log level from warn to v3

If a pod has a configmap/secret volume an annoying message shows up
in the log approximately every 70 seconds. This happens because the
desiredStateOfWorldPopulator sync loop always call the
MarkRemountRequired. The function finds the volume plugin and check
if the plugin requires mount. Configmap and secret plugins always
returns true for that. Thus, the reconciler code of the volume manager
remounts the volume every time. This commit decrease the log level of
that message in the mount function from warning to V4.

Signed-off-by: José Guilherme Vanz <jguilhermevanz@suse.com>

MikeSpreitzer committed:
Replaced repair with returning error, in delta_fifo.go

When dedupDeltas does the impossible and the key is already queued,
return an error rather than maintain the data structure invariants.

yuanchen8911 committed:
Fix a typo in PreBindPlugin comment

 "before a pod is being scheduled"  ->" before a pod is bound"

maryamklabib committed:
Resolves staticcheck failures for component-base/metrics

gaurav1086 committed:
TestCreateInvokeAdmissionControl: remove unnecessary goroutine in sequential processing

Signed-off-by: Gaurav Singh <gaurav1086@gmail.com>

Commit:
[pkg/controller/certificates]: remove staled func comments

This was introduced by commit: f04ce3cfba6ca20ca9e8b71caf65bdb391d70030
Since this func is simple and clear enough, just not comment it anymore.

Signed-off-by: Zhou Peng <p@ctriple.cn>

nikhita committed:
apimachinery: remove inactive members from OWNERS

nikhita committed:
test/images/pets: remove OWNERS to cleanup inactive members

bprashanth hasn't be active since the release of v1.11. Removing them
from test/images/pets/OWNERS would leave mkumatag as the sole approver.

But mkumatag is also an approver for test/images/OWNERS so this commit
removes the test/images/pets/OWNERS completely.

Note: we should try to find more OWNERS for test/images/pets instead,
but this cleanup is a short term solution to avoid the bot suggesting
inactive members for reviews and approval.

nikhita committed:
cluster/addons/metadata-proxy: cleanup inactive members from OWNERS

andyzhangx committed:
chore: add diskclient.ListByResourceGroup interface

update bazel config

gnufied committed:
Remove resize map code

This code is not used anywhere

andyzhangx committed:
chore: add diskclient.Update interface

Commit:
Move WatchUntilWithoutRetry() from e2e framework

WatchUntilWithoutRetry() is called from e2e/apps/rc.go only.
So this moves the function to reduce e2e/framework/util.go code.

Commit:
Move MasterUpgrade() out from e2e framework

MasterUpgrade() is called only from cloud/gcp/cluster_upgrade.go.
And the function depends on GCP, so it is nice to move this function
from e2e framework.

Commit:
Minor cleanup in snapshot test suite.

Commit:
Add pre provisioned snapshot tests

Make pre provisioned snapshots using CSI driver by
1. Take a dynamic snapshot with retain policy
2. Delete the dynamic snapshot and content
3. Create a preprovisioned snapshot with snapshotHandle

This commit adds a preprovisiond test pattern, all snapshots made using
create snapshot resource become prepv snapshots. All exisitng test cases
now run again with prepv snapshots.

Huang-Wei committed:
remove duplicate path import

yiduyangyi committed:
fix golint failures in pkg/kubeapiserver/options

lixiaobing1 committed:
fix HandleCrash() order

msau42 committed:
Export WaitForCSIDriverRegistrationOnAllNodes to be used by external csi driver repos

Change-Id: Ie61430b1050a778d8ba98177e0c995ff2553f9cd

gaurav1086 committed:
[kubelet/dockershim] : Use local copy for range iterator

Signed-off-by: Gaurav Singh <gaurav1086@gmail.com>

abhiraut committed:
Fix NetworkPolicy describe for egress-all policies

NetworkPolicy egress-all `kubectl describe` output should refer
to the term "destination" as opposed to "source" for describing
policies which do not restrict traffic based on the destination.

Commit:
fix(staticcheck): fix static error of package `/pkg/volume/gcepd`

Signed-off-by: Dong Gang <dong.gang@daocloud.io>

julianvmodesto committed:
Add tests for managed fields tracking.

- Test that client-side apply users don't encounter a conflict with
server-side apply for objects that previously didn't track managedFields
- Test that we stop tracking managed fields with `managedFields: []`
- Test that we stop tracking managed fields when the feature is disabled

roycaihw committed:
use canonical import path in client-gen so that package comments are

honored

roycaihw committed:
add vendorless to client-gen

roycaihw committed:
generated

soltysh committed:
Use temporary directory when creating temporary file in tests

FreeZhang61 committed:
add GetAddressAndDialer unit test

Signed-off-by: zhangsheng02 <sheng.zhang@daocloud.io>

Commit:
Combine two tests with overlap to reduce overall snapshot test suite time

Commit:
add bootstrap policy for monitoring roles

(we enable metrics and pprof by default, but that doesn't mean
 we should have full cluster-admin access to use those endpoints)

Change-Id: I20cf1a0c817ffe3b7fb8e5d3967f804dc063ab03

remove pprof but add read access to detailed health checks

Change-Id: I96c0997be2a538aa8c689dea25026bba638d6e7d

add base health check endpoints and remove the todo for flowcontrol, as there is an existing ticket

Change-Id: I8a7d6debeaf91e06d8ace3cb2bd04d71ef3e68a9

drop blank line

Change-Id: I691e72e9dee3cf7276c725a12207d64db88f4651

lixiaobing1 committed:
remove some notes about scheduler/algorithm

jvanz committed:
Revert "cleanup: decrease log level from warn to v3"

This reverts commit ebece49936e635f151fdd8a64fa2b77fd183e817.

yuzhiquan committed:
Fixed error string should not be capitalized

yuzhiquan committed:
Fixed potential nil dereference in hostpath unit test

jvanz committed:
cleanup: print warning message after timeout

If a pod has a configmap/secret volume an annoying message shows up
in the log approximately every 70 seconds. This happens because the
desiredStateOfWorldPopulator sync loop always call the
MarkRemountRequired. The function finds the volume plugin and check
if the plugin requires mount. Configmap and secret plugins always
returns true for that. Thus, the reconciler code of the volume manager
remounts the volume every time. This commit change the SetVolumeOwnership
to print the warning only if the function does not finish within 30
seconds.

Signed-off-by: José Guilherme Vanz <jguilhermevanz@suse.com>

liggitt committed:
Handle int -> float conversion in FromUnstructured

philjb committed:
AWS: increase io1 volume IOPs limit

Max IOPS for SSD (io1) volumes was increased from 20,000 to 32,000:
https://aws.amazon.com/about-aws/whats-new/2017/12/amazon-ebs-provisioned-iops-ssd--io1--volumes-now-support-32-000-iops-and-500-mbs-per-volume/

and later to 64,000:
https://aws.amazon.com/about-aws/whats-new/2018/11/amazon-elastic-block-store-announces-double-the-performance-of-provisioned-iops-volumes/

runzexia committed:
clean up unused var containerCache

runzexia committed:
update

dixudx committed:
remove serializers from codec factory

runzexia committed:
update build

runzexia committed:
regen

andyzhangx committed:
fix: azure disk resize error if source does not exist

fix gofmt

yue9944882 committed:
apf integration test: ensure no rejection

yiduyangyi committed:
fix golint failures in pkg/kubeapiserver/options, use API Server in commemts instead of APIServer

yiduyangyi committed:
fix golint failures in pkg/kubeapiserver/options, rename receiver name of BuiltInAuthorizationOptions to o

yiduyangyi committed:
fix golint failures in pkg/kubeapiserver/options, fix some incorrect replace of receiver name

fisherxu committed:
return err directly when nodename is not consistent in cert

Commit:
Remove snapshot class from prepv test because it is not needed.

Prepv snapshots only need to specify driver.

prameshj committed:
Delete ILB FR in case of changes to port/proto.

prameshj committed:
unit test

liggitt committed:
Shorten watch restart test, run in parallel

Commit:
add more testing cases

Signed-off-by: zhangsheng02 <sheng.zhang@daocloud.io>

MikeSpreitzer committed:
Remove phantoms from dump_requests output

No longer writes a single info-free line for each exempt priority level.

nikhita committed:
hack: ensure go version is not specified for master in publishing rules

The go version for all master branches is defined in `default-go-version`
locally.

Note that this validation isn't added to publishing-bot directly because
we still want to give users of the publishing-bot the ability to specify
specific go versions for the master branch if they need, but we want to
explicitly disallow this for kubernetes repos.

lemonli committed:
nodelocaldns.yaml: remove force_tcp option which used for external dns query

Commit:
Remove false positive warning in kubeadm cmd

zshihang committed:
bind metadata proxy to 0.0.0.0

gongguan committed:
extend ShouldCallHook benchmark to verify performance imporvement

Commit:
Update k8s.io/utils dependency and use ebtables from there

ZeroMagic committed:
chore: upgrade Azure/go-autorest to v14.2.0
use autorest v0.11.1

Signed-off-by: ZeroMagic <jiliu8@microsoft.com>

Commit:
Quick commit

t0rr3sp3dr0 committed:
Use NLB Subnet CIDRs instead of VPC CIDRs in updateInstanceSecurityGroupsForNLB

Signed-off-by: Pedro Tôrres <t0rr3sp3dr0@gmail.com>

wojtek-t committed:
Fix verbs reporting in kube-apiserver metrics

Commit:
reduce type conversion and correct variable name

smarterclayton committed:
kubectl: Allow --patch-file to be passed to `kubectl patch`

The current structure is limited in the size of the patch it can
accept depending on the shell of the environment (for instance, a
900kb patch cannot be applied in bash 4 on Fedora 32). Allow a user
to specify `--patch-file` to provide a file containing the patch
rather than directly on the command line.

claudiubelu committed:
tests: Fixes variable expansion false positive test

Some of the tests are negative test cases which are supposed to ensure that those
invalid usecases are handled properly.

However, some of the tests are false positives, they can pass due to various reasons.
One such example is: "should fail substituting values in a volume subpath with absolute path".
This test can pass if:

- the Pod cannot start due to various reasons (e.g.: the container image cannot be pulled or does
  not exist).
- the Pod ran to completion, even though the container was not supposed to start in the first place.

liggitt committed:
Clean up daemonset test names

SergeyKanzhelev committed:
remove legacy leftovers of portmapping functionality that was moved to CNI

SergeyKanzhelev committed:
revert auto-completed package names

vikkyomkar committed:
removed k8s-master label

sbangari committed:
fix the remote endpoint cleanup logic

chendave committed:
Add chendave to sig-scheduling REVIEWERS

gobomb committed:
let panics propagate up when processLoop panic

Mr-Linus committed:
Remove unnecessary conversion

Update framework.go

Update framework.go

Update framework.go

remove unnecessary conversion

remove unnecessary conversion

remove unnecessary conversion

remove unnecessary conversion

lesszyao committed:
Cleanup wait forever loops in pv_controller_test.go

kshitijcode committed:
Added support for register-with-taints setting #93608

gireeshpunathil committed:
fix a typo in the comment

kshitijcode commented:
/assign @mm4tt

mm4tt commented on cmd/kubemark/hollow-node.go:
```diff
@@ -88,6 +89,9 @@ func (c *hollowNodeConfig) addFlags(fs *pflag.FlagSet) {
 	fs.Var(&bindableNodeLabels, "node-labels", "Additional node labels")
+	bindableRegisterWithTaints := cliflag.ConfigurationMap(c.RegisterWithTaints)
+	fs.Var(&bindableRegisterWithTaints, "register-with-taints", "Adding Taints")
+	
```
Remove blank line

mm4tt commented on cmd/kubemark/hollow-node.go:
```diff
@@ -64,6 +64,7 @@ type hollowNodeConfig struct {
 	ProxierSyncPeriod    time.Duration
 	ProxierMinSyncPeriod time.Duration
 	NodeLabels           map[string]string
+	RegisterWithTaints   map[string]string
```
You also need to propagate this to the kubelet flags, exactly like the NodeLabels is done - https://github.com/kubernetes/kubernetes/blob/8d54a84507c622e1552373d5d125c84ee22d1465/pkg/kubemark/hollow_kubelet.go#L164

kshitijcode replied:
@mm4tt resolved :)

kshitijcode replied:
@mm4tt : I changed it to type []core.Taint
Is it as intended?

kshitijcode commented:
/assign @mm4tt

neolit123 commented:
/sig scalability
are release notes needed for kubemark?

thtanaka committed:
kubeadm: reset don't unmount /var/lib/kubelet if it is mounted

Previous PR that fixes this
https://github.com/kubernetes/kubernetes/pull/71663

Signed-off-by: Thomas Tanaka <thomas.tanaka@gmail.com>

SergeyKanzhelev committed:
PodReasonUnschedulable is not a pod condition type

houminz committed:
Fix an error in NodeUnschedulable plugin comment

kshitijcode commented:
/help 
Sorry, I am unaware but what are the next steps on this PR workflow?
I tried searching for the relevant documentation but couldn't find .

Jiawei0227 committed:
Detect change of volume attachability in the middle of attaching

- Add Unit tests for both volumemanager and attach/detach controller
- Add E2E test

mm4tt commented on cmd/kubemark/hollow-node.go:
```diff
@@ -88,6 +89,8 @@ func (c *hollowNodeConfig) addFlags(fs *pflag.FlagSet) {
 	bindableNodeLabels := cliflag.ConfigurationMap(c.NodeLabels)
 	fs.Var(&bindableNodeLabels, "node-labels", "Additional node labels")
+	bindableRegisterWithTaints := cliflag.ConfigurationMap(c.RegisterWithTaints)
+	fs.Var(&bindableRegisterWithTaints, "register-with-taints", "Adding Taints")
```
Please change the help message to something like: "Register the hollow node with the given list of taints"

mm4tt commented:
We don't need release notes for kubemark, please set them to none

/ok-to-test

kshitijcode commented:
/retest

kshitijcode commented:
@mm4tt : I am not able to figure out what failed the tests. It doesn't seem to be related to my change.
Can you please confirm?

kshitijcode commented:
@mm4tt : I am not sure about the reason for failures. They don't seem to be related to my change. Can you please confirm?

neolit123 commented:
/release-note-none

kshitijcode commented:
/retest

kshitijcode commented:
/retest

kshitijcode commented:
@mm4tt 
What should be the default value when exposing RegisterWithTaints in &KubeletFlags

```return &KubeletFlags{
		ContainerRuntimeOptions: *NewContainerRuntimeOptions(),
		CertDirectory:           "/var/lib/kubelet/pki",
		RootDirectory:           defaultRootDir,
		MasterServiceNamespace:  metav1.NamespaceDefault,
		MaxContainerCount:       -1,
		MaxPerPodContainerCount: 1,
		MinimumGCAge:            metav1.Duration{Duration: 0},
		NonMasqueradeCIDR:       "10.0.0.0/8",
		RegisterSchedulable:     true,
		RemoteRuntimeEndpoint:   remoteRuntimeEndpoint,
		NodeLabels:              make(map[string]string),
		RegisterNode:            true,
		SeccompProfileRoot:      filepath.Join(defaultRootDir, "seccomp"),
		// prior to the introduction of this flag, there was a hardcoded cap of 50 images
		EnableCAdvisorJSONEndpoints: false,
		RegisterWithTaints: 	 []core.Taint,
	}```

ynqa committed:
add testcases for kubelet getters

kshitijcode commented:
/help

Commit:
benchmark-dockerized.sh: set KUBE_ROOT

See https://testgrid.k8s.io/sig-scalability-benchmarks#scheduler for an
example of this script failing.

wking committed:
leaderelection: Remove doubled space in "attempting to acquire leader lease" log

The extra space was introduced in 5b854e7b17 (say which lease is being
acquired, 2018-01-16, #58341).

Commit:
Stack snapshot tests into one to reuse snapshot resource and reduce time taken on prow.

mrkm4ntr committed:
Use v1helper.GetPersistentVolumeClass for compatibility

gongguan commented:
RegisterWithTaints is a slice not a map, you should replace `bindableRegisterWithTaints := cliflag.ConfigurationMap(c.RegisterWithTaints)` like this
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/options/options.go#L338

Commit:
Remove custom hash function, generate a random UUID for snapshot and snapcontent name instead

kshitijcode commented:
@mm4tt @gongguan 
I have made the necessary change. I am not sure why the tests are failing. Locally it seems to be working fine.

gongguan commented:
pass RegisterWithTaints from config to options here
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubemark/hollow-node.go#L109

kshitijcode commented:
> https://github.com/kubernetes/kubernetes/blob/master/cmd/kubemark/hollow-node.go#L109

@gongguan I added it there.

What about adding it here as well https://github.com/kubernetes/kubernetes/blob/382107e6c84374b229e6188207ef026621286aa2/cmd/kubelet/app/options/options.go#L191 ?

kshitijcode commented:
> pass RegisterWithTaints from config to options here
> https://github.com/kubernetes/kubernetes/blob/master/cmd/kubemark/hollow-node.go#L109

Also, how do we interpret the failed tests, some of the tests just say the failed error without much information?

gongguan commented:
you should format changed files and run hack/update-bazel.sh

kshitijcode commented:
@gongguan I did it. 
`hack/update-bazel.sh` didn't spit out an error.

mm4tt commented:
The build is failing with:
```
W0806 08:27:38.879] Use --sandbox_debug to see verbose messages from the sandbox
W0806 08:27:38.880] compilepkg: missing strict dependencies:
W0806 08:27:38.880] 	/bazel-scratch/.cache/bazel/_bazel_prow/48d5366022b4e3197674c8d6e2bee219/sandbox/linux-sandbox/6785/execroot/io_k8s_kubernetes/pkg/kubemark/hollow_kubelet.go: import of "k8s.io/kubernetes/pkg/apis/core"
```

Yo need to add the deps to the bazel's BUILD files

kshitijcode commented:
@mm4tt Do you mean adding 
``` "//pkg/apis/core:go_default_library",
        "//pkg/apis/core/pods:go_default_library",
        "//pkg/apis/core/v1:go_default_library",
        "//pkg/apis/core/v1/helper:go_default_library",
        "//pkg/apis/core/v1/helper/qos:go_default_library",```
to https://github.com/kubernetes/kubernetes/blob/master/pkg/kubemark/BUILD

kshitijcode commented:
@mm4tt : added the dependencies, not really surely if its at the right place.

gongguan commented:
> @mm4tt : added the dependencies, not really surely if its at the right place.
> Still doesn't seem to be fixed .

Run `./hack/update-bazel.sh`, BUILD files will be corrected automatically.

kshitijcode commented:
/test pull-kubernetes-e2e-kind-ipv6

kshitijcode commented:
@mm4tt @gongguan I fixed most of the errors. Some tests seem to failing.  Not really sure what the exact root cause is.

Commit:
Move common portion of dynamic snapshot out of switch statement

gongguan commented:
> @mm4tt @gongguan I fixed most of the errors. Some tests seem to failing. Not really sure what the exact root cause is.

I think these are flaky tests which not related to your pr, try retest to succeed them.
You should squash your commits into one at first because it's misleading in git logs.

kshitijcode commented:
/retest pull-kubernetes-integration

kshitijcode commented:
/test pull-kubernetes-verify

kshitijcode commented:
/test pull-kubernetes-integration

kshitijcode commented:
@mm4tt Can you please help me with a `lgtm` ?

knight42 committed:
fix(kubelet): protect `containerCleanupInfos` from concurrent map writes

Signed-off-by: knight42 <anonymousknight96@gmail.com>

knight42 committed:
test(kubelet): add a regression test to verify kubelet would not panic

Commit:
Ran hack/update-bazel.sh but discarded changes to root build file

lesszyao committed:
Seperate registry and feature gates

habibrosyad committed:
fix test/e2e_node staticcheck

andyzhangx committed:
fix: smb remount issue

typo

typo

add one logging

jsafrane committed:
Fix waiting for PVCs to get Bound

When Get() returns error, the loop should not return with nil. Make sure
that the faulty Get() marks phaseFoundInAllClaims as not satisfied.

jsafrane committed:
Fix Poll variable name

Parameter names should start with lowercase

jsafrane committed:
Fix error messages

Separate different errors by ": ".

dims commented:
/assign @wojtek-t

mm4tt commented:
Sorry, this fell off my radar.

/lgtm
/approve

kshitijcode commented:
@mm4tt : What would be the next steps to get this merged?

mm4tt commented:
It will automatically get merged once the 1.19 code thaw is lifted. 
It will land into 1.20 release. The change is not critical enough to be part of 1.19

alexzimmer96 committed:
Refactored pkg/controllers/resourcequota

* Fixed golint issues
* Removed redundant package import names
* Improved some variables and names

Co-authored-by: Mike Danese <mikedanese@gmail.com>

ESWZY committed:
Add test for `pkg/kubelet/util/util_windows_test.go#GetAddressAndDialer`

michaelgugino committed:
Handle eviction of pods in deleted namespace

If a pod is already marked deleted, and the eviction
api returns an unauthorized response, ignore that
error since the pod is marked for deletion already.

If the pod is not already marked deleted, retry.

pjbgf committed:
Update yaml files to use seccomp GA syntax

phiphi282 committed:
Allow 404 error on lb deletion in azure

When trying to delete a loadbalancer after all resources in azure
including the resource group have been deleted we currently get an an
error when trying to delete the loadbalancer afterwards because the
resource group hasn't been found.

Change EnsureLoadBalancerDeleted function to not fail when the resource
group has already been deleted.

dcbw committed:
proxy/userspace: use waitgroups instead of sketchy atomic ops in testcases

Instead of relying on atomic ops to increment/decrement at the right time
just use waitgroups to provide hard synchronization points.

MHBauer committed:
update test to match validation filter of pods

invalid pod filter changed to name only

changed for k8s v1.9 in 811447ea0a9928c326965f3bbd8195826b20daf8 PR#53194

andyzhangx committed:
fix: return error with fewer mount options on Windows

vareti committed:
start kube-apiserver and webhook server only once to shorten the webhook audit test time

habibrosyad committed:
fix vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructuredscheme staticcheck

habibrosyad committed:
fix vendor/k8s.io/apimachinery/pkg/conversion staticcheck

soulxu committed:
Only lock statuses map when status isn't successful

Determine the status successful or not first, that will reduce the chance
of lock contention.

claudiubelu committed:
tests: Use Windows absolute path on Windows pods

The test "should fail substituting values in a volume subpath with absolute path" creates a pod
with a variable expansion path which is set as an absolute path. However "/tmp" is not an absolute
on Windows, it has to be prefixed with the drive letter (C:\tmp). But C:\tmp does not typically
exist on Windows nodes, so we use C:\Users instead.

ynqa committed:
delete tests which use container manager stub

giuseppe committed:
kubelet, cgroupv2: do not create /sys/fs/cgroup/sys with cgroupfs

Closes: https://github.com/kubernetes/kubernetes/issues/94104

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

robscott committed:
Updating kube-proxy to trim space from loadBalancerSourceRanges

Before this fix, a Service with a loadBalancerSourceRange value that
included a space would cause kube-proxy to crashloop. This updates
kube-proxy to trim any space from that field.

andyzhangx committed:
fix: use sensitiveOptions on Windows mount

tkashem committed:
Make similar buckets for api and etcd request duration histogram

Make similar buckets for the apiserver_request_duration_seconds and
the etcd_request_duration_seconds histogram so that the result is
more comparable side by side.

etcd_request_duration_seconds uses the default buckets provided by
prometheus client library:
DefBuckets = []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10}

apiserver_request_duration_seconds on the other hand uses more fine
grained buckets, and the maximum bucket size is 60s. Both histograms
should use similar bucket sizes so they are more comparable side by side.

RyPeck committed:
Make ProjectedVolumeSource optional

tkashem committed:
Add impersonated user to system:authenticated group

Currently if a group is specified for an impersonated user,
'system:authenticated' is not added to the 'Groups' list inside the
request context.
This causes priority and fairness match to fail. The catch-all flow
schema needs the user to be in the 'system:authenticated' or in the
'system:unauthenticated' group. An impersonated user with a specified
group is in neither.

As a general rule, if an impersonated user has passed authorization
checks, we should consider him authenticated.

Commit:
[pkg/registry/admissionregistration]: fixup wrong copy&paste godoc

This was introduced at commit: fc5a613c17c81fdd

Signed-off-by: Zhou Peng <p@ctriple.cn>

feiskyer committed:
Ensure backoff step is set to 1 for Azure armclient

mattfenwick committed:
modify DNS-1123 error messages to indicate that RFC 1123 is not followed exactly

soulxu committed:
Initialize candidate directly instead of iterating the array of candidates

Using existing victimsMap to get the victims, then it is easy to build candidate
directly.

derekwaynecarr committed:
Verify running mirror pod has running containers

javidiaz committed:
Adding label NodeAlphaFeature to include tests in Node Testgrid

kevindelgado committed:
Add exp backoff for connection refused errors

Currently when ListAndWatch() receives a connection refused error, it is
assumed to be due to the apiserver being transiently unresponsive. In
situations where a controller is running outside the k8s cluster it's
controlling, it is more common for the controller to lose connection
permanently to the apiserver and needs to exponentially backoff its
retry rather than continously spamming logs with Watch attempts that
will never succeed.

jingxu97 committed:
Set snapshotType for tests with NTFS.

Set the snapshotType for tests with NTFS. Return error if type is not
set.

jingxu97 committed:
Remove file_mode args for Windows test

Due to issue https://github.com/kubernetes/kubernetes/issues/94237,
remove file_mode arg for subPath tests for now.

deads2k committed:
update drain message to include namespace for pod in kubectl get compatible format

jingyih committed:
Update etcd client to v3.4.13

RainbowMango committed:
remove deprecated kubelet endpoint /metrics/resource/v1alpha1

dims committed:
Move brendandburns to to emeritus status.

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

fedepaol committed:
Mark SCTPConnectivity tests as disruptive.

Since the SCTP module verification tests were added, their result may be affected by
running the SCTPConnectivity tests. For this reason, they are now marked as disruptive.

Signed-off-by: Federico Paolinelli <fpaoline@redhat.com>

Mr-Linus committed:
Update options.go

jingyih committed:
Update default etcd server to 3.4.13

Commit:
Combine switch case into one case

SergeyKanzhelev committed:
fix linter issues for pkg/kubelet/cri/remote and staging/src/k8s.io/cri-api/pkg/apis/testing

millermatt committed:
Remove duplicate nodeSelector

Make the yaml valid.

knight42 committed:
fix(azure): check error returned by scaleSet.getVMSS

Signed-off-by: knight42 <anonymousknight96@gmail.com>

xmudrii committed:
Update cri-tools to v1.19.0

javidiaz committed:
Moving e2e tests to e2e_node directory

liggitt committed:
Deprioritize extensions/v1beta1 in discovery

Commit:
Add a link to liggitt's deflake docs in Flake template (#94243)

* Add a link to liggitt's deflake docs in template

We could have prow bot link comment the link but I think this should be enough to increase visibility to the doc when people encouter a flake

* Update with more standard docs

Commit:
Fix testpattern check

janeczku committed:
Fixed reflector not recovering from "Too large resource version" errors with API servers 1.17.0-1.18.5

kevindelgado committed:
Fix FakeClock::Reset to always succeed

Current FakeClock::Reset only successfully resets the timer and
returns true when the timer has neither fired nore been stopped.

This is incorrect behavior as real clocks allow for reseting
in both of those situations (as long has the channel has been drained).

This is useful in tests that use a fake clock to test
wait.BackofManager, as the timer resets after each iteration of
Backoff() after the timer has already fired.

liggitt committed:
Improve docs for client-go warning handlers

bbyrne5 committed:
fix staticcheck errors in resourcequota

ialidzhikov committed:
Update max data disk count with new instance types

Signed-off-by: ialidzhikov <i.alidjikov@gmail.com>

liggitt committed:
Remove 1.17.0 API testdata

liggitt committed:
Add 1.19.0 API testdata

joremysh committed:
fixed golint error in pkg/apis/admissionregistration

hs0210 committed:
Improve the prompt message when kubeadm init is successful

Signed-off-by: Hu Shuai <hus.fnst@cn.fujitsu.com>

feiskyer committed:
Ensure getPrimaryInterfaceID not panic when network interfaces for Azure VMSS are null

joremysh committed:
fix golint failures

some packages under staging/src/k8s.io/sample-apiserver/

kshitijcode commented:
@mm4tt : I see the thaw is lifted. Can this be merged now for the subsequent release?

wojtek-t commented:
It will be picked automatically. The code thaw is not fully lifted - we're lifting that in phases.

justaugustus committed:
Update CNI plugins to v0.8.7

ref: https://github.com/containernetworking/plugins/releases/tag/v0.8.7

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

mikkeloscar committed:
Omit optional field when empty

Annotates the optional field `StabilizationWindowSeconds` with
`omitempty` such that it will be omitted when creating resources where
the field is not set.

Signed-off-by: Mikkel Oscar Lyderik Larsen <mikkel.larsen@zalando.de>

MHBauer committed:
relax matcher for ResourceMetricsAPI test

Instead of requiring only exactly the pods created, allow other pods
to exist by ignoring any extra matches.

jingxu97 committed:
Start CSI proxy as service

Now CSI proxy can support starting as service. Update node startup
script to create and start it as service

mkumatag committed:
Skip TestLoopbackHostPortIPv6 run on non-IPv6 env

loopever committed:
fix typos in cmd/kubeadm

rosti committed:
kubeadm: Ensure etcd manifests are regenerated upon upgrade

A check that verifies that kubeadm does not "upgrade" to an older release was
overly optimized by skipping upgrade if the new version is the same as the old
one. This somewhat makes sense, but that way changes in any of the etcd fields
in the ClusterConfiguration won't be applied if the etcd version is not
changed.

Hence, this simple change ensures that the upgrade is done even when no version
change takes place.

Signed-off-by: Rostislav M. Georgiev <rostislavg@vmware.com>

neolit123 committed:
kubeadm: make the scheduler and KCM connect to local endpoint

Pinning the kube-controller-manager and kube-scheduler kubeconfig files
to point to the control-plane-endpoint can be problematic during
immutable upgrades if one of these components ends up contacting an N-1
kube-apiserver:
https://kubernetes.io/docs/setup/release/version-skew-policy/#kube-controller-manager-kube-scheduler-and-cloud-controller-manager
For example, the components can send a request for a non-existing API
version.

Instead of using the CPE for these components, use the LocalAPIEndpoint.
This guarantees that the components would talk to the local
kube-apiserver, which should be the same version, unless the user
explicitly patched manifests.

lavalamp committed:
move port definitions to a common location

lavalamp committed:
fix doc file

brahmaroutu committed:
moving files from k8s.io/util/mount into staging/src/k8s.io/mount-utils

rosti committed:
kubeadm: Fix `upgrade plan` for air-gapped setups

A bug was discovered in the `enforceRequirements` func for `upgrade plan`.
If a command line argument that specifies the target Kubernetes version is
supplied, the returned `ClusterConfiguration` by `enforceRequirements` will
have its `KubernetesVersion` field set to the new version.
If no version was specified, the returned `KubernetesVersion` points to the
currently installed one.

This remained undetected for a couple of reasons
- It's only `upgrade plan` that allows for the version command line argument to
  be optional (in `upgrade plan` it's mandatory)
- Prior to 1.19, the implementation of `upgrade plan` did not make use of the
  `KubernetesVersion` returned by `enforceRequirements`.

`upgrade plan` supports this optional command line argument to enable
air-gapped setups (as not specifying a version on the command line will end up
looking for the latest version over the Interned).

Hence, the only option is to make `enforceRequirements` consistent in the
`upgrade plan` case and always return the currently installed version in the
`KubernetesVersion` field.

Signed-off-by: Rostislav M. Georgiev <rostislavg@vmware.com>

Commit:
CHANGELOG: Update directory for v1.16.15 release

lavalamp committed:
partial

dcbw committed:
proxy/userspace: clean up and consolidate testcase setup

lavalamp committed:
rename

lavalamp committed:
update scripts

lavalamp committed:
fix a few missed references

```
grep -RIl pkg/master . | xargs -n 1 sed -i 's|pkg/master|pkg/controlplane|g'
LC_ALL=C sort -o hack/.golint_failures hack/.golint_failures
```

mikedanese committed:
remove --verbose_failures from .bazelrc

wojtek-t commented:
@kshitijcode - please rebase the PR (we're ready to merge it)

wojtek-t commented:
Something didn't work - there are now 250 commits in it..

kshitijcode commented:
Checking ..

kshitijcode commented:
@wojtek-t  I am not able to figure out just did a git-rebase. Can you help me with the same?

wojtek-t commented:
I'm not able to fix your PR - you need to do it yourself.

kshitijcode committed:
Merge branch 'fix-for-issue-93608' of github.com:kshitijcode/kubernetes into fix-for-issue-93608

kshitijcode commented:
@wojtek-t Sorry, I think something went wrong with my git history- reforked and  raised a fresh PR.
